includes/Pages/PageRequestFormManagement.php

```diff
@@ -0,0 +1,313 @@
+                $proceed = false;
+            }
+
+            if ($form->getOverrideQueue() !== null) {
+                /** @var RequestQueue|bool $queue */
+                $queue = RequestQueue::getById($form->getOverrideQueue(), $database);
+                if ($queue === false || $queue->getDomain() !== $domainId || !$queue->isEnabled()) {
+                    SessionAlert::error("The chosen queue does not exist or is disabled.");
+                    $proceed = false;
+                }
+            }
+
+            if ($proceed) {
+                $form->save();
+                Logger::requestFormCreated($database, $form);
+                $this->redirect('requestFormManagement');
+            }
+            else {
+                $this->populateFromObject($form);
+                WebRequest::setSessionContext('preview', [
+                    'main' => $form->getFormContent(),
+                    'username' => $form->getUsernameHelp(),
+                    'email' => $form->getEmailHelp(),
+                    'comment' => $form->getCommentHelp(),
+                ]);
+
+                $this->assign('createMode', true);
+                $this->setTemplate('form-management/edit.tpl');
+            }
+        }
+        else {
+            $this->populateFromObject(new RequestForm());
+            WebRequest::setSessionContext('preview', null);
+            $this->assign('hidePreview', true);
+
+            $this->assignCSRFToken();
+            $this->assign('createMode', true);
+            $this->setTemplate('form-management/edit.tpl');
+        }
+    }
+
+    protected function view()
+    {
+        $database = $this->getDatabase();
+
+        /** @var RequestForm $form */
+        $form = RequestForm::getById(WebRequest::getInt('form'), $database);
+
+        if ($form->getDomain() !== Domain::getCurrent($database)->getId()) {
+            throw new AccessDeniedException();
+        }
+
+        $this->populateFromObject($form);
+
+        if ($form->getOverrideQueue() !== null) {
+            $this->assign('queueObject', RequestQueue::getById($form->getOverrideQueue(), $database));
+        }
+
+        WebRequest::setSessionContext('preview', [
+            'main' => $form->getFormContent(),
+            'username' => $form->getUsernameHelp(),
+            'email' => $form->getEmailHelp(),
+            'comment' => $form->getCommentHelp(),
+        ]);
+
+        $renderer = new MarkdownRenderingHelper();
+        $this->assign('renderedContent', $renderer->doRender($form->getFormContent()));
+
+        $this->setTemplate('form-management/view.tpl');
+    }
+
+    protected function edit()
+    {
+        $database = $this->getDatabase();
+
+        /** @var RequestForm $form */
+        $form = RequestForm::getById(WebRequest::getInt('form'), $database);
+
+        if ($form->getDomain() !== Domain::getCurrent($database)->getId()) {
+            throw new AccessDeniedException();
+        }
```

Comment:
This check is to make sure the user's current domain is able to see this specific object. It's basically one of the checks I wanted to add in #600 (see your other comment about the FIXMEs).

Since this is a new object, `create()` doesn't need this check as the object will be *created* in the user's current domain (see L81).

Since preview never actually calls the database and just populates from what's in the user's session, there's no object to check against. The user *must* have been able to see an object (or create an object) already for this to work.